docs/reference/query-dsl/bool-query.asciidoc

```diff
@@ -60,6 +60,18 @@ POST _search
 }
 --------------------------------------------------
 
+[[bool-min-should-match]]
+==== Using `minimum_should_match`
+
+You can use the `minimum_should_match` parameter to specify the number or
+percentage of `should` clauses returned documents _must_ match.
+
+If the `bool` query includes only `should` clauses, the default value is `1`.
+If the `bool` query includes any other type of clause, the default value is `0`.
```

Comment:
It doesn't cover the case when a `bool` query includes `should` and `must_not` clauses, when the default value would still be 1. The below description would be closer to how we choose the default internally (I wish it was not that complicated).

```
If the `bool` query includes at least one `should` clause and no `must` or `filter` clauses, the default value is `1`.
Otherwise, the default value is `0`.
```

Reply:
Thanks for the clarification @jpountz. I've updated this section as you suggested with 35a41bf2004dff086fca19e0e3a67c6da651133a.